py/mpconfig.h

```diff
@@ -603,6 +603,11 @@
 #define MICROPY_ENABLE_EXTERNAL_IMPORT (MICROPY_CONFIG_ROM_LEVEL_AT_LEAST_CORE_FEATURES)
 #endif
 
+// Whether to enable parsing __all__ when importing all public symbols from modules
+#ifndef MICROPY_ENABLE_MODULE_ALL
```

Comment:
```suggestion
#ifndef MICROPY_ENABLE_MODULE_IMPORT_ALL
```

perhaps a more informative name?

Reply:
Agreed, I have changed it accordingly

Reply:
I think the config name needs to start with `MICROPY_MODULE_`, because that's the existing prefix used for module/package/import configuration, eg `MICROPY_MODULE_GETATTR` enables support for module-level `__getattr__`.

So maybe `MICROPY_MODULE_ALL`?  That does seem a bit obscure, so it could be `MICROPY_MODULE___ALL__` (following `MICROPY_PY___FILE__`) or `MICROPY_MODULE_IMPORT_ALL`